### PR TITLE
hardened soles quirk

### PR DESCRIPTION
**Pull request by Seris02**
## About The Pull Request

adds a +2 positive quirk that gives you the TRAIT_HARDSOLES trait which stops you from being affected by things that happen with shoes off.

## Why It's Good For The Game

I dunno, thought it would be a good quirk, plus WFWTP asked for it a while ago, and 

tbh I can't think of anything else so _shrug_

## Changelog
:cl:
add: hardened soles trait
/:cl:

### CODE REVIEW

**Comment by ghost** [comment]
😂

**Comment by Trilbyspaceclone** [comment]
maybe give it to robo legs/ash walker legs?

**Review by Ghommie — CHANGES_REQUESTED** [review]
Powergaming anti-caltrop quirk. It has to be tweaked before even being accepted, ergo, only stop knockdowns when walking and halve caltrop damage, not prevent it.

**Comment by Seris02** [comment]
ah, alrighty

**Comment by necromanceranne** [comment]
This is actually a powergame trait since being shoeless is what happens when you lose a leg.

Traits were never meant to be combat based and this is definitely partially combat based given a piece of glass underfoot of a one legged opponent (or one lacking shoes) is a totally viable tactic made irrelevant by this trait.

**Comment by Seris02** [comment]
are you saying skittish isn't

File: code/datums/components/footstep.dm [hunk 1]
```diff
@@ -89,7 +89,7 @@
 			return
 		if(ishuman(LM)) //for proper humans, they're special
 			var/mob/living/carbon/human/H = LM
-			var/feetCover = (H.wear_suit && (H.wear_suit.body_parts_covered & FEET)) || (H.w_uniform && (H.w_uniform.body_parts_covered & FEET) || (H.shoes && (H.shoes.body_parts_covered & FEET)))
+			var/feetCover = HAS_TRAIT(H, TRAIT_HARDSOLES) || (H.wear_suit && (H.wear_suit.body_parts_covered & FEET)) || (H.w_uniform && (H.w_uniform.body_parts_covered & FEET) || (H.shoes && (H.shoes.body_parts_covered & FEET)))
```
Comment (Linzolle) [inline code/datums/components/footstep.dm]: why would this be here..? This makes it so the footstep noise won't play if you have the trait, even if you're barefoot and it should play the barefoot noise. Are feet shoes?

**Comment by Owai-Seek** [comment]
You're saying someone who's lost a leg isn't already at a severe disadvantage?

Anyway, as a resident footwrap wearing lizard, I don't deal with antagonist stuff that often. The -only- time I've ever dealt with things related to a lack of footwear is when I step on glass or D4's. Punji sticks don't care if you're wearing shoes or not, and I've only walked on those a few times for daring to enter botany.

File: code/datums/components/footstep.dm (hunk 1 above)
Comment (Seris02) [reply code/datums/components/footstep.dm]: oof, sorry, was just looking for one specific piece of code and kinda replacing it mindlessly

File: code/datums/components/caltrop.dm [hunk 2]
```diff
@@ -46,7 +46,9 @@
 			return
 
 		var/damage = rand(min_damage, max_damage)
-		if(HAS_TRAIT(H, TRAIT_LIGHT_STEP))
+		if (HAS_TRAIT(H,TRAIT_HARDSOLES))
```
Comment (Ghommie) [inline code/datums/components/caltrop.dm:49]: Add a variable so you won't run the macro twice in this procedure.

File: code/modules/assembly/mousetrap.dm [hunk 3]
```diff
@@ -46,7 +46,7 @@
 			return FALSE
 		switch(type)
 			if("feet")
-				if(!H.shoes || !(H.shoes.body_parts_covered & FEET))
+				if(!HAS_TRAIT(H,TRAIT_HARDSOLES) && (!H.shoes || !(H.shoes.body_parts_covered & FEET)))
```
Comment (Ghommie) [inline code/modules/assembly/mousetrap.dm:49]: move the trait check two lines below, as well as add another walk move_intent check on that line.

File: code/datums/components/caltrop.dm [hunk 4]
```diff
@@ -59,4 +61,5 @@
 						"<span class='userdanger'>You slide on [A]!</span>")
 
 			cooldown = world.time
-		H.Knockdown(60)
+		if (!HAS_TRAIT(H,TRAIT_HARDSOLES))
```
Comment (Ghommie) [inline code/datums/components/caltrop.dm:64]: Walk move_intent check, please.

File: code/datums/traits/good.dm [hunk 5]
```diff
@@ -219,3 +219,9 @@
 /datum/quirk/night_vision/on_spawn()
 	var/mob/living/carbon/human/H = quirk_holder
 	H.update_sight()
+
+/datum/quirk/hardsoles
+	name = "Hardened Soles"
+	desc = "The soles of your feet are hardened from walking barefoot for so long. You no longer have to worry about stepping on anything, sharp or otherwise."
```
Comment (Ghommie) [inline code/datums/traits/good.dm:225]: Guess you'll have to edit the description a little after applying the suggestions.

**Review by Ghommie — CHANGES_REQUESTED** [review]

**Comment by Weblure** [comment]
> This is actually a powergame trait since being shoeless is what happens when you lose a leg.
> 
> Traits were never meant to be combat based and this is definitely partially combat based given a piece of glass underfoot of a one legged opponent (or one lacking shoes) is a totally viable tactic made irrelevant by this trait.

Plenty of traits can be and are used to aid in combat (walk faster, see in the dark, dive into locked lockers, hold more blood, heal from alcohol, free mood boost, etc.). This is honestly one of the least useful traits for powergaming. Shoes can easily be found all over the station, and you can even spawn with jackboots, which can discreetly hold items. There's absolutely no good reason to not wear shoes other than for RP purposes. Nobody is going to waste 2 points on this over more useful quirks if they're looking for an edge in combat.

**Comment by silicons** [comment]
Blocking ALL caltrops is an absolute no.
We already have enough insanely poweragmey traits (like night vision which for the record I don't like that well but people will fucking screech if I hammer it back to /tg/ levels but I wouldn't be against that) we don't need more.

**Comment by silicons** [comment]
Also I'm going to be honest this goes for everyone:
If most of the reasoning is "someone asked me to" I'm going to start dragging the person who asked up to provide their reasoning and use it and that is probably going to result in a close half the time because of what their reasoning tends to be. 
PLEASE. PUT. ACTUAL. REASONING. IN. if you want this stuff to be taken seriously.

**Pull request closed without merging**